doc/dev/adr/0027-generic-usage-of-cloud-event-headers.md

```diff
@@ -0,0 +1,51 @@
+# ADR27: Generic usage of cloud event headers
+
+## Context: 
+
+> Note that this ADR builds upon previous ADRs [10](./0010-cloud-event-content-type.md) and [11](./0011-cloud-events-api.md).
+
+
+There have been asks for our SDK to support the creation and parsing of cloud event headers in a generic enough way that any application could take it as a dependency even if that application doesn't use our telemetry/RPC protocol clients.
+
+Currently, our protocol package does allow users to pass in cloud event headers when sending telemetry messages, but the knowledge of what MQTT user property name is associated with each cloud event header is hidden within the protocol library.
+
+## Decision: 
+
+In each of our language protocol libraries, we will expose two new functions:
+
+ 1. One for parsing a "cloud event" type from a given set of MQTT user properties
+ 2. One for populating a list of MQTT user properties from a given "cloud event" type.
+   - This user flow should include the SDK providing default values for ```type``` and ```specversion``` of "ms.aio.telemetry" and "1.0" respectively, but should not provide any other default values.
```

Comment:
should we also provide "default" values for id (new generated guid) and time (current timestamp)?